pkg/provisioner/ironic/ironic.go

```diff
@@ -1162,21 +1162,21 @@ func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareCompone
 	if !p.availableFeatures.HasFirmwareUpdates() {
 		return nil, fmt.Errorf("current ironic version does not support firmware updates")
 	}
+
+	// Setting to 2 since we only support bmc and bios
+	componentsInfo := make([]metal3api.FirmwareComponentStatus, 0, 2)
```

Comment:
The inefficiency of writing the simpler `componentsInfo := []metal3api.FirmwareComponentsStatus{}` (or `var componentsInfo []metal3api.FirmwareComponentsStatus`) here is dwarfed by the inefficiency of storing wholes structs instead of pointers in the array, so all the data has to be copied every time it is passed around.